File: leetcodePython/math/942.di-string-match.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        solution = []
        start = 0
        end = len(S)
        for i in S:
            if i == 'I':
                solution.append(start)
                start += 1
            else:
                solution.append(end)
                end -= 1
        solution.append(start)
        return solution

File: leetcodeJS/string/942.di-string-match.js
/**
 * @param {string} S
 * @return {number[]}
 */
var diStringMatch = function (S) {
  if (S.length === 0) return [];
  let d = S.length;
  let i = 0;
  let solution = [];
  for (let j = 0; j < S.length + 1; j++) {
    if (S[j] === "D") {
      solution.push(d);
      d--;
    } else {
      solution.push(i);
      i++;
    }
  }
  return solution;
};
